src/function/gds/output_writer.cpp

```diff
@@ -36,9 +37,24 @@ std::unique_ptr<common::ValueVector> RJOutputWriter::createVector(const LogicalT
     return vector;
 }
 
+void RJOutputWriter::beginWritingForDstNodesInTable(common::table_id_t tableID) {
+    rjOutputs->beginWritingOutputsForDstNodesInTable(tableID);
+    if (outputNodeMask != nullptr) {
```

Comment:
Is it easy/possible to get rid of outputNodeMask != nullptr and always ensure its not null (but a default one that always returns true?)? That would make the code cleaner and safer I think.

Reply:
I think this would be solved once #4356 is merged.